discovery.go

```diff
@@ -292,5 +306,8 @@ func (d *Server) outputProcessor(outWriter io.Writer) {
 			}
 			fmt.Fprintln(outWriter, string(data))
 		}
+		// We finished consuming all messages, now
+		// we can exit for real
+		d.outputWaiter.Done()
```

Comment:
What about dropping the channel + goroutine + waitgroup + `d.quit` in change of a mutex?

```
outputMutext.Lock()
output.Write(data)
outputMutex.Unlock()
```

Reply:
That would be mean refactor everything again, I'd prefer to avoid that. :confounded: